src/fu-util.c

```diff
@@ -234,6 +236,11 @@ fu_util_update_device_changed_cb (FwupdClient *client,
 	else if (fwupd_device_has_flag (device, FWUPD_DEVICE_FLAG_NEEDS_REBOOT))
 		priv->completion_flags |= FWUPD_DEVICE_FLAG_NEEDS_REBOOT;
 
+	if (priv->current_message == NULL) {
+		message = fwupd_device_get_update_message (priv->current_device);
+		if (message != NULL)
+			priv->current_message = message;
```

Comment:
This makes memory management somewhat tricky. Why not just g_strdup it rather than hoping `priv->current_device` stays alive?

Reply:
That does sound safer.  I'll set that up and `g_free` both in the common finalize as well as when it's displayed though.

Otherwise updating more than one CAB file will mean only one message can be displayed.

Reply:
> Otherwise updating more than one CAB file will mean only one message can be displayed

This is the niggle at he back of my head; what if there is more than one message for the device? For composite devices like a hub that's not outside the realm of possibility. I'm wondering if we actually want fwupd_device_add_update_message() and fwup_device_get_update_messages(), or maybe we just want to query each child for messages in the client.

Reply:
I had that same thought - but then also was thinking is it really "useful" to a customer to see more than 1 message on a composite device?  I would think it makes sense to either:

* put the message on a single device that causes the update (that's how I think I would use this)
or
* put the same message on all devices and have the client only show one for the composite devices.